src/python/pants/backend/python/subsystems/debugpy.py

```diff
@@ -34,6 +34,22 @@ class DebugPy(PythonToolBase):
 
     args = ArgsListOption(example="--log-to-stderr")
 
+    @property
+    def all_requirements(self) -> tuple[str, ...]:
+        """All the raw requirement strings to install the tool.
+
+        This may not include transitive dependencies: these are top-level requirements.
+        """
+        # NB: Locking with importlib_metadata is only necessary while debugpy supports Py 3.7, as
+        # `importlib.metadata` was added in Py 3.8 and would allow us to resolve the console_script
+        # using just the stdlib.
+        #
+        # Lastly, importlib_metadata, has so-far maintained a backwards-compatible API for
```

Comment:
This whole paragraph seems like over-explanation. As far as I'm aware all backport libs like this maintain stdlib compatibility and stdlib doesn't break people.

Reply:
Thanks for the link. To the contrary IIUC. Today it returns `SelectableGroups.load(...).select()` which is an `EntryPoints`, in the future `EntryPoints` directly. Either way, the API we us on the returned `EntryPoints` is iteration. `EntryPoint` derives from a `DeprecatedList` which is what provides the iteration over contained `EntryPoint` objects, but the iterability is not the deprecated thing. Finally, we use 2 APIs from the iterated `EntryPoint` objects, `ep.name` and `ep.load()`, and neither are going away.

Reply:
> Either way, the API we use on the returned EntryPoints is iteration

Not true, We use the `SelectableGroups` dict-interface (the `['console_scripts']` at the end of the second line.

Reply:
Ah, yes. Super confusing. Nonetheless, EntryPoints also supports a dict-interface and so both paths allow the current code to work. That said, the EntryPoints dict-interface is deprecated: https://github.com/python/importlib_metadata/blob/051b26a890a717dd9aa59fcbbb039b55a57c0372/importlib_metadata/__init__.py#L359-L374

Quoting myself:
> As far as I'm aware all backport libs like this maintain stdlib compatibility and stdlib doesn't break people.

And that is wrong! Right on the tin in intro: https://docs.python.org/3.8/library/importlib.metadata.html
> Note
> 
> This functionality is provisional and may deviate from the usual version semantics of the standard library.

So the real issue here doesn't seem to be importlib_metadata so much as importlib.metadata. I was not aware Python allowed itself to break people now within a major version.

Sorry for the thrash.